### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by Tails128**
Updated the README.md for the following reasons:

- Added a better separation for Gradle and Maven
- Added an explanation to use `properties` in Maven, for beginner-friendliness
- Added links to verify what the latest versions of the plugins are
- Separated Java and Kotlin GRPC versions to avoid issues
- Specified to add `os-maven-plugin` as it's a prerequisite

### CODE REVIEW

File: compiler/README.md [hunk 1]
```diff
@@ -52,70 +54,118 @@ protobuf {
...
+    <kotlin.version>1.5.31</kotlin.version> <!-- you should already have this line -->
+
+    <!- the version is the lower of the ones found in these two links
+    (they should always be synced, but double checking is better than being stuck on an error):
+    https://search.maven.org/search?q=a:protoc-gen-grpc-kotlin
+    https://search.maven.org/search?q=a:grpc-kotlin-stub -->
+    <grpc.kotlin.version>1.2.0</grpc.kotlin.version>
+
+    <!-- The version is the latest found here: https://search.maven.org/artifact/io.grpc/grpc-protobuf/1.42.0/jar -->
+    <java.grpc.version>1.42.0</java.grpc.version>
```
Comment (jamesward) [inline compiler/README.md:75]: I think the max we currently support is `1.39.0`

File: compiler/README.md [hunk 2]
```diff
@@ -52,70 +54,118 @@ protobuf {
...
+
+    <!-- the version is the latest found here: https://search.maven.org/search?q=a:protobuf-kotlin -->
+    <protobuf.version>3.19.1</protobuf.version>
```
Comment (jamesward) [inline compiler/README.md:78]: I think the max we currently support is `3.18.1`

File: compiler/README.md [hunk 3]
```diff
@@ -52,70 +54,118 @@ protobuf {
...
+        <version>0.6.1</version>
+        <executions>
+            <execution>
+                <id>compile</id>
+                <goals>
+                    <goal>compile</goal>
+                    <goal>compile-custom</goal>
+                </goals>
+                <configuration>
+                    <protocArtifact>com.google.protobuf:protoc:YOUR_PROTOBUF_VERSION:exe:${os.detected.classifier}</protocArtifact>
```
Comment (jamesward) [inline compiler/README.md:138]: Use property here

File: compiler/README.md [hunk 4]
```diff
@@ -52,70 +54,118 @@ protobuf {
...
+                    <pluginId>grpc-java</pluginId>
+                    <pluginArtifact>io.grpc:protoc-gen-grpc-java:YOUR_GRPC_VERSION:exe:${os.detected.classifier}</pluginArtifact>
```
Comment (jamesward) [inline compiler/README.md:140]: Use property here

File: compiler/README.md [hunk 5]
```diff
@@ -52,70 +54,118 @@ protobuf {
...
+                    <protocPlugins>
+                        <protocPlugin>
+                            <id>grpc-kotlin</id>
+                            <groupId>io.grpc</groupId>
+                            <artifactId>protoc-gen-grpc-kotlin</artifactId>
+                            <version>YOUR_GRPC_KOTLIN_VERSION</version>
```
Comment (jamesward) [inline compiler/README.md:146]: Use property here

File: compiler/README.md [hunk 6]
```diff
@@ -52,70 +54,118 @@ protobuf {
...
+                    </protocPlugins>
+                </configuration>
+            </execution>
+            <execution>
+                <id>compile-kt</id>
+                <goals>
+                    <goal>compile-custom</goal>
+                </goals>
+                <configuration>
+                    <protocArtifact>com.google.protobuf:protoc:YOUR_PROTOBUF_VERSION:exe:${os.detected.classifier}</protocArtifact>
```
Comment (jamesward) [inline compiler/README.md:159]: Use property here

File: compiler/README.md (hunk 1 above)
Comment (Tails128) [reply compiler/README.md:75]: Is there a place where there's track of the maximum supported version for this and `protobuf-kotlin` ?

File: compiler/README.md (hunk 3 above)
Comment (Tails128) [reply compiler/README.md:138]: for the os.detected.classifier?

File: compiler/README.md (hunk 1 above)
Comment (jamesward) [reply compiler/README.md:75]: I don't know of a good resource for that currently.  Ultimately we need to get gRPC Kotlin upgraded to work with the latest versions: https://github.com/grpc/grpc-kotlin/issues/289

File: compiler/README.md (hunk 3 above)
Comment (jamesward) [reply compiler/README.md:138]: For `YOUR_PROTOBUF_VERSION`

Comment (Tails128) [reply compiler/README.md:138]: oh! pardon me, forgot to change it "^^

File: compiler/README.md (hunk 1 above)
Comment (Tails128) [reply compiler/README.md]: Should be done, would you prefer me to resolve for this and the following points

**Review by jamesward — APPROVED** [review]

**Comment by Tails128** [comment]
Thanks for the approval 🎉

Just in case I do more contributions, is it fine if I close the PR on my own after the checks pass, or should I let it to the reviewers?

**Comment by jamesward** [comment]
Thanks @Tails128! I was waiting for the tests to pass and thanks for bringing this back to my attention. Merged!

**Pull request merged**